platform/stm32f4xx/irq_manager.cpp

```diff
@@ -0,0 +1,90 @@
+#include <platform/irq_manager.hpp>
+#include <new>
+
+IRQ_manager::handler_storage IRQ_manager::m_storage[IRQ_manager::irqs];
+
+
+void IRQ_manager::init()
+{
+    // Initialize storage
+    for (auto &h : m_storage) {
+        new (&h) handler_type{default_handler};
+    }
+
+    __enable_irq();
+}
+
+int IRQ_manager::subscribe(IRQn_t irqn, const std::function< void() > &handler)
+{
```

Comment:
irqn can be negative (as IRQn_t defined in stm32f4xx.h). Would be nice to add a check at least.

Does this also mean that I can set a custom handler for first 15 interrupts (say UsageFault_IRQn) ?

The void IRQ_manager::isr() says it is possible, since we subtract 16 from IPSR value. So, 6 is actually UsageFault, and its handler should be at handlers[6]